src/compiler/types.ts

```diff
@@ -1525,7 +1526,9 @@ module ts {
     export interface SymbolLinks {
         target?: Symbol;                    // Resolved (non-alias) target of an alias
         type?: Type;                        // Type of value symbol
-        declaredType?: Type;                // Type of class, interface, enum, or type parameter
+        declaredType?: Type;                // Type of class, interface, enum, type alias, or type parameter
+        typeParameters?: TypeParameter[];   // Type parameters of type alias (empty array if none)
+        instantiations?: Map<Type>;         // Instantiations of generic type alias

```

Comment:
Does it make sense to move the `typeParameters` and `instantiations` here for classes and interfaces as well? Seems like it would be nice if they were all in one place.